### ISSUE TITLE
have a question about the candlelight_stm32f4_fw project.

### ISSUE THREAD

**Issue by cheoljun99** [lucaszfolle/candlelight_stm32f4_fw#1]
I am currently working on a CAN Sniffer implementation project and I am impressed by your project that implements the CandleLight with the STM32f4 board.

I have flashed your firmware code exactly to the STM32F407G-disc1 board and entered the command
sudo ip link set can0 type can bitrate 500000
sudo ip link set can0 up
in Linux and then the cansend command, but
the red LED blinks 3 times and then no transmission occurs.

Also, I cannot send and receive CAN data with the CAN interface that has the same 500kps transmission rate.

I connected the CAN transceiver to PD0 and PD1 and connected the CAN transceiver to the CAN bus, but there is no response.
Does the type of transceiver affect it? Also, do I need a 120 ohm resistor?

The issue has been resolved now.

The CAN TX, RX of PD0 and PD1 was the problem. I set the CAN TX, RX to a different port and tried again, and it worked.

**Comment by lucaszfolle** [comment]
EDIT: nvm, I saw your edit just now. I'm glad you solved it, congrats!

Hi Park, it's been a couple of years since the last time I tinkered with this firmware. But I'll give it a try.
So if I understood correctly, the eval board is recognized as the can0 interface and you were able to activate the interface correctly, otherwise the red and blue LEDs would not be on. Right?

> Does the type of transceiver affect it? Also, do I need a 120 ohm resistor?

I used the SN65HVD1050, but any transceiver would do the job. As for the resistor, the bus should have one 120 ohm resistor on each end, making the desired bus load of 60 ohm. On a short bus for test purposes it will work with a wide range of resistor values, but it absolutely needs a resistance load to communicate.

Did you test the communication between two discovery eval boards or with a different device? Try the command 'ifconfig can0' to see if it shows any RX/TX packet count. Check if CAN H and L are correctly connected and try measuring the voltage across them and to the GND, to make sure it stays normally on the recessive state.

Keep me informed if you have any progress.
